Mjml.Net/Components/Body/CarouselComponent.cs

```diff
@@ -0,0 +1,453 @@
+        }
+
+        private void GenerateControls(IHtmlRenderer renderer, string direction, string icon)
+        {
+            var (iconWidth, iconUnit) = UnitParser.Parse(IconWidth);
+
+            renderer.StartElement("td") // Style controls.td
+                .Class($"mj-carousel-{CarouselID}-icons-cell")
+                .Style("font-size", "0px")
+                .Style("display", "none")
+                .Style("mso-hide", "all")
+                .Style("padding", "0px");
+
+            renderer.StartElement("div") // Style controls.div
+                .Class($"mj-carousel-{direction}-icons")
+                .Style("display", "none")
+                .Style("mso-hide", "all");
+
+            for (int i = 0; i < CarouselImages.Count(); i++)
+            {
+                var image = CarouselImages.ElementAt(i);
+
+                renderer.StartElement("label")
+                    .Attr("for", $"mj-carousel-{CarouselID}-radio-{i}")
+                    .Class($"mj-carousel-{direction}")
+                    .Class($"mj-carousel-{direction}-{i}");
+
+                renderer.StartElement("img") // Style controls.img
+                    .Attr("src", icon)
+                    .Attr("alt", direction)
+                    .Attr("width", $"{iconWidth}")
+                    .Style("display", "block")
+                    .Style("height", "auto")
+                    .Style("width", IconWidth);
+
+                renderer.EndElement("img");
+                renderer.EndElement("label");
+            }
+
+            renderer.EndElement("div");
+            renderer.EndElement("td");
+        }
+
+        private void GenerateImages(IHtmlRenderer renderer, GlobalContext context)
+        {
+            renderer.StartElement("td") // Style images.td
+                .Style("padding", "0px");
+
+            renderer.StartElement("div")
+                .Class("mj-carousel-images");
+
+            for (int i = 0; i < CarouselImages.Count(); i++)
+            {
+                var image = CarouselImages.ElementAt(i);
+
+                image.CarouselID = CarouselID;
+                image.CarouselImageIndex = i;
+                image.BorderRadius = BorderRadius;
+
+                image.Render(renderer, context);
+            }
+
+            renderer.EndElement("div");
+            renderer.EndElement("td");
+        }
+
+        private void RenderFallback(IHtmlRenderer renderer, GlobalContext context)
+        {
+            var firstImage = CarouselImages.FirstOrDefault();
+
+            if (firstImage != null)
+            {
+                renderer.StartConditional("<!--[if mso]>");
+                {
+                    firstImage.BorderRadius = BorderRadius;
+                    firstImage.Render(renderer, context);
+                }
+                renderer.EndConditional("<![endif]-->");
+            }
+        }
+
+        private void HeadStyle(IHtmlRenderer renderer, GlobalContext context)
```

Comment:
@SebastianStehle This is horrible. Any ideas on how we could better approach this?